src/projectscene/view/toolbars/audiosetupcontextmenumodel.cpp

```diff
@@ -0,0 +1,152 @@
+        makeMenu(muse::TranslatableString("audio setup", "Recording device"), makeRecordingDevicesItems(), "recordingDeviceMenu"),
+        makeMenu(muse::TranslatableString("audio setup", "Recording channels"), makeInputChannelsItems(), "inputChannelsMenu"),
+        makeMenuItem("rescan-devices"),
+        makeMenuItem("audio-settings")
+    };
+
+    setItems(items);
+}
+
+MenuItemList AudioSetupContextMenuModel::makeHostItems()
+{
+    MenuItemList items;
+    auto currentApi = audioDevicesProvider()->currentAudioApi();
+
+    auto makeChangeApiAction = [](int index) {
+        ActionQuery q("action://playback/change-api");
+        q.addParam("api_index", muse::Val(index));
+        return q;
+    };
+
+    const auto& apiList = audioDevicesProvider()->audioApiList();
+    for (int i = 0; i < apiList.size(); ++i) {
+        MenuItem* item = makeMenuItem(makeChangeApiAction(i).toString(),
+                                      muse::TranslatableString("audio setup", muse::String::fromStdString(apiList.at(i))));
+        item->setId(QString::fromStdString(item->query().toString()));
+        if (currentApi == apiList.at(i)) {
+            item->setChecked(true);
+        }
+        items << item;
+    }
+
+    return items;
+}
+
+MenuItemList AudioSetupContextMenuModel::makePlaybackDevicesItems()
+{
+    MenuItemList items;
+    auto currentOutputDevice = audioDevicesProvider()->currentAudioOutputDevice();
+
+    auto makeChangePlaybackDeviceAction = [](int index) {
+        ActionQuery q("action://playback/change-playback-device");
+        q.addParam("device_index", muse::Val(index));
+        return q;
+    };
+
+    const auto& outputDevicesList = audioDevicesProvider()->audioOutputDevices();
+    for (int i = 0; i < outputDevicesList.size(); ++i) {
+        MenuItem* item = makeMenuItem(makeChangePlaybackDeviceAction(i).toString(),
+                                      muse::TranslatableString("audio setup", muse::String::fromStdString(outputDevicesList.at(i))));
+        item->setId(QString::fromStdString(item->query().toString()));
+        if (currentOutputDevice == outputDevicesList.at(i)) {
+            item->setChecked(true);
+        }
+        items << item;
+    }
+
+    return items;
+}
+
+MenuItemList AudioSetupContextMenuModel::makeRecordingDevicesItems()
+{
+    MenuItemList items;
+    auto currentInputDevice = audioDevicesProvider()->currentAudioInputDevice();
+
+    auto makeChangeRecordingDeviceAction = [](int index) {
+        ActionQuery q("action://playback/change-recording-device");
+        q.addParam("device_index", muse::Val(index));
+        return q;
+    };
+
+    const auto& inputDevicesList = audioDevicesProvider()->audioInputDevices();
+    for (int i = 0; i < inputDevicesList.size(); ++i) {
+        MenuItem* item = makeMenuItem(makeChangeRecordingDeviceAction(i).toString(),
+                                      muse::TranslatableString("audio setup", muse::String::fromStdString(inputDevicesList.at(i))));
```

Comment:
need to use `TranslatableString::untranslatable(muse::String::fromStdString(inputDevicesList.at(i))))`  instead of `muse::TranslatableString("audio setup", muse::String::fromStdString([inputDevicesList.at](http://inputdeviceslist.at/)(i))))`